### PR TITLE
feat(im): add opt-in normalized message list JSON

### PR DESCRIPTION
**Pull request by xiaoxiangyu-123**
## Summary

Add an explicit normalized JSON shape for `im +chat-messages-list` and `im +threads-messages-list`. This is an independent evaluation of the compact-existing-JSON approach and does not depend on or modify #2567.

Default JSON remains backward compatible. Pass `--json-shape normalized` to hoist repeated `chat_id` / `thread_id` context and sender metadata into `participants`, with messages referencing reusable senders through `sender_id`.

## Changes

- Add `--json-shape legacy|normalized` to both existing message-list shortcuts; the default is `legacy`.
- Make jq filter the shape selected by `--json-shape`.
- Keep `pretty`, `table`, `csv`, and `ndjson` on the established per-message projection. Unknown-format JSON fallback also stays legacy unless normalization was explicitly requested.
- Preserve inline `sender` objects when a sender has no stable ID or the same ID has conflicting metadata.
- Normalize nested `thread_replies` without dropping opaque values, enriched fields, or mismatched per-message context.
- Document the opt-in schema and participant lookup in the IM affordance and shipped skill references.

API requests, pagination behavior, reactions, thread expansion, resource downloads, message-body enrichment, and default output contracts are unchanged.

## Test Plan

- [x] `go test ./shortcuts/im ./internal/affordance -count=1`
- [x] `go test ./tests/cli_e2e/im -run TestIM_ListPageAllDryRun -count=1`
- [x] `node scripts/skill-format-check/index.js`
- [x] `make vet`
- [x] `make fmt-check`
- [x] `make build`
- [x] `QUALITY_GATE_CHANGED_FROM=origin/main make quality-gate`
- [x] `go mod tidy` leaves `go.mod` and `go.sum` unchanged
- [ ] Full local `make unit-test`: affected packages pass; unrelated subprocess tests timed out under the full race run, then passed in isolation. CI is authoritative.
- [ ] Live IM E2E: attempted, but the available bot fixture returned OpenAPI code 230027 for message-list reads. Existing and new dry-run/unit coverage passed.

## Related Issues

- Alternative implementation to #2567; that PR is intentionally unchanged.

## Summary by CodeRabbit

* **New Features**
  * Added `--json-shape` to chat and thread message-list commands.
  * Preserved legacy JSON output by default, with normalized JSON available via `--json-shape normalized`.
  * Normalized output reduces repeated conversation and sender data while retaining necessary context.
  * Non-JSON formats and filtering behavior remain unchanged.

* **Documentation**
  * Updated command references and tips with JSON shape details and usage examples.

* **Tests**
  * Added coverage for legacy and normalized output, filtering, pagination, and edge cases.

### CODE REVIEW

File: shortcuts/im/message_compact.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+	return out
+}
+
+func commonMessageString(messages []map[string]interface{}, key string) string {
+	common := ""
+	conflict := false
+	walkMessageTree(messages, func(message map[string]interface{}) {
+		value, _ := message[key].(string)
+		if value == "" || conflict {
+			return
```
[the remark this answers is not in the document]
Comment (xiaoxiangyu-123) [reply shortcuts/im/message_compact.go]: Fixed in 69690e49: missing, empty, or non-string chat_id values now prevent hoisting, and regression coverage verifies both missing and non-string cases.

File: internal/affordance/im_source_test.go [hunk 2]
```diff
@@ -166,6 +166,19 @@ func TestIMAffordanceDoesNotDuplicateRuntimeRecovery(t *testing.T) {
...
 
+func TestIMMessageListAffordanceDocumentsNormalizedJSON(t *testing.T) {
+	prev := mdSource
+	t.Cleanup(func() { SetSource(prev) })
+	SetSource(os.DirFS("../../affordance"))
+
+	for _, method := range []string{"+chat-messages-list", "+threads-messages-list"} {
+		tips := parsedIMAffordance(t, method).Tips
+		if !containsItem(tips, "participants") || !containsItem(tips, "sender_id") {
+			t.Errorf("%s tips must explain normalized participant lookup: %v", method, tips)
```
[the remark this answers is not in the document]
Comment (xiaoxiangyu-123) [reply internal/affordance/im_source_test.go]: Fixed in 69690e49: the affordance source test now asserts chat_id for +chat-messages-list and thread_id for +threads-messages-list, in addition to the explicit opt-in and participant lookup.

File: internal/affordance/im_source_test.go [hunk 3]
```diff
@@ -166,6 +166,19 @@ func TestIMAffordanceDoesNotDuplicateRuntimeRecovery(t *testing.T) {
 	}
 }
 
+func TestIMMessageListAffordanceDocumentsNormalizedJSON(t *testing.T) {
+	prev := mdSource
+	t.Cleanup(func() { SetSource(prev) })
+	SetSource(os.DirFS("../../affordance"))
```
[the remark this answers is not in the document]
Comment (xiaoxiangyu-123) [reply internal/affordance/im_source_test.go]: Addressed in 69690e49 with a narrow test-only //nolint:forbidigo annotation. SetSource requires fs.FS, and the path is a bounded repository-local fixture; this matches the surrounding test setup without changing production filesystem boundaries.
